### PR TITLE
Add coverage and move CoffeeScript tests to TS

### PR DESCRIPTION
**Pull request by Jarlakxen**
I want to contribute this changes because I think that could help to contributors to spot places to improve. Because most of the proyect is in TS, I moved CoffeeScript tests to TS. For the coverga I use this proyect as a guide https://github.com/Izhaki/Typescript-Jasmine-Istanbul-Boilerplate.
There are some minor changes in the code, most of them are just small refactors. I tried to keep the same logic and only change code styles.

### CODE REVIEW

File: src/spec/swank-extras.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  it('should parse scala notes', () => {
+    const input = `
+        (:scala-notes (:is-full nil :notes ((:file "/Users/viktor/dev/projects/kostbevakningen/src/main/scala/se/kostbevakningen/model/record/Ingredient.scala" :msg "missing
+        arguments for method test in object Ingredient; follow this method with \`_' if you want to treat it as a partially applied function" :severity error :beg 4138 :end 4142 :line 105 :col 3))))
+    `
+
+    const result = sexpToJObject(readFromString(input))
+    expect(result[':scala-notes'][':notes'].length).toBe(1)
+  })
+})
```
Comment (hedefalk) [inline src/spec/swank-extras.spec.ts]: This entire file could be killed off, it's just old swank protocol things which aren't used anymore…

**Review by hedefalk — CHANGES_REQUESTED** [review]
Thanks, great!

Haven't run the tests yet but I think I'll just merge first. If you could just remove that dead swank file, that'd be great!

(I'm also trying out the review functionality of github so want to request change just for the heck of it, sorry :)

**Comment by hedefalk** [comment]
Oh, I also noticed that travis failed, could you look at that? Might be broken before though, but I would really like a working ci…

**Comment by Jarlakxen** [comment]
I just checked Travis, the test are failing due to the node version. It look like that ts-node only works with node >= 6 and the test are running with 4 and 5.

File: src/spec/swank-extras.spec.ts (hunk 1 above)
Comment (Jarlakxen) [reply src/spec/swank-extras.spec.ts]: Sure! No problem. Only the spec? or also the swank-extras.ts?

**Comment by hedefalk** [comment]
Ah, ok. I'm all for latest and greatest, but do we really need ts-node? (Had googled what it was)

Isn't it better to just run tests from compiled js?

**Comment by hedefalk** [comment]
I think it might still be needed to parse the .ensime file but that test
definitely is "dead code"

**Comment by Jarlakxen** [comment]
I'm using ts-node in a very similar way to https://babeljs.io/docs/usage/babel-register/. I choosed ts-node because I found it much easier to integrate with Istanbul. I think, if we choose to compile the files, we need to use https://github.com/SitePen/remap-istanbul to remap Istanbul code coverage information to its original source. I didn't found a good example of remap-istanbul + jasmine + TS.
But I'm open to other solutions =D

**Comment by Jarlakxen** [comment]
Any thoughts? =)

**Comment by hedefalk** [comment]
@Jarlakxen Sorry I haven't answered, been super busy…

I'm open to suggestions, just want that green tick basically :)

**Review by hedefalk — APPROVED** [review]

**Comment by hedefalk** [comment]
Thx!

**Comment by Jarlakxen** [comment]
No problem! 
I the meantime, I was working on other PRs =D

**Comment by hedefalk** [comment]
Great! They are much welcome!

**Pull request merged**